File: dart/bogosort2.dart
bool isSorted(List<int> array) {
  for (var i = 1, len = array.length; i < len; i++) {
    if (array[i-1] > array[i]) {
      return false;
    }
  }
  return true;
}

void main() {
  var array = [0, 3, 2, 4, 10, 1, 7, 9];
  while (!isSorted(array)) {
    array.shuffle();
  }
  print("$array");
}

File: rust/bogosort2.rs
extern crate rand;
use rand::{seq::SliceRandom, thread_rng};

fn is_sorted(arr: &[i32]) -> bool {
    for i in 1..arr.len() {
        if arr[i-1] >= arr[i] {
            return false;
        }
    }
    true
}

fn main() {
    let mut arr: [i32; 8] = [10, 2, 7, 4, 6, 1, 9, 0];
    while !is_sorted(&arr) {
        arr.shuffle(&mut thread_rng());
    }
    println!("{:?}", arr);
}
